File: eLobato/src/PrimeFactors.rb
require 'mathn'

class PrimeFactors
 def self.generate(int)
   solution = Array.new
   if(int.is_a? Integer)
    for number in 1..int do
      if (int % number == 0) and (number.prime?)
        solution << number
      end
    end
   else
    false
   end
   return solution
 end
end

File: fiunchinho/PrimeFactors.php
<?php
/**
 * Description of Bowling: http://butunclebob.com/ArticleS.UncleBob.ThePrimeFactorsKata
 *
 * @author fiunchinho
 */
class PrimeFactors {

	/**
	 * Try to factor the a number.
	 *
	 * @param int $number Number that we want to get the factors from.
	 * @return array Array containing the prime factors.
	 */
	public static function generate( $number )
	{
		for( $result = array(), $divider = 2; $number >= 2; $divider++ )
		{
			while( $number % $divider == 0 )
			{
				array_push( $result, $divider ) ;
				$number = (int) $number / $divider ;
			}
		}
		return $result;
	}
}
?>
